src/box/gc.c

```diff
@@ -913,11 +920,16 @@ gc_erase_consumer(const struct tt_uuid *uuid)
 		 tt_uuid_str(uuid)) != 0)
 		return -1;
 
-	/* Do actual work on commit. */
-	struct trigger *on_commit =
-		xregion_alloc_object(&in_txn()->region, struct trigger);
-	trigger_create(on_commit, gc_erase_consumer_on_commit, consumer, NULL);
-	txn_stmt_on_commit(txn_current_stmt(in_txn()), on_commit);
+	if (in_txn() != NULL) {
+		/* Do actual work on commit. */
+		struct trigger *on_commit =
+			xregion_alloc_object(&in_txn()->region, struct trigger);
+		trigger_create(on_commit, gc_erase_consumer_on_commit, consumer,
+			       NULL);
+		txn_stmt_on_commit(txn_current_stmt(in_txn()), on_commit);
+	} else {
+		gc_erase_consumer_finalize(consumer);
```

Comment:
That doesn't look right. The change is not transactional. There is time when the replica is already not in `_gc_consumers`, but in-memory its consumer still exists. Please, wrap `boxk()` into a txn when there isn't one existing already. Then you can leave the remaining code untouched. I.e. always erase on commit.

Reply:
Fixed - now I always delete the in-memory consumer in an on_commit trigger.